src/System.Device.Gpio/System/Device/Pwm/PwmController.cs

```diff
@@ -1,114 +1,128 @@
+        /// </summary>
+        /// <param name="driver">The driver that manages all of the channel operations for the controller.</param>
+        public PwmController(PwmDriver driver)
+        {
+            _driver = driver;
+            _openChannels = new HashSet<(int, int)>();
+        }
+
+        /// <summary>
+        /// Opens a channel in order for it to be ready to use.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip.</param>
+        /// <param name="pwmChannel">The PWM channel.</param>
+        public void OpenChannel(int pwmChip, int pwmChannel)
+        {
+            if (_openChannels.Contains((pwmChip, pwmChannel)))
+            {
+                throw new InvalidOperationException("The selected channel is already open.");
+            }
+
+            _driver.OpenChannel(pwmChip, pwmChannel);
+            _openChannels.Add((pwmChip, pwmChannel));
+        }
+
+        /// <summary>
+        /// Closes an open channel.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip.</param>
+        /// <param name="pwmChannel">The PWM channel.</param>
+        public void CloseChannel(int pwmChip, int pwmChannel)
+        {
+            if (!_openChannels.Contains((pwmChip, pwmChannel)))
+            {
+                throw new InvalidOperationException("Can not close a channel that is not open.");
+            }
+
+            _driver.CloseChannel(pwmChip, pwmChannel);
+            _openChannels.Remove((pwmChip, pwmChannel));
+        }
+
+        /// <summary>
+        /// Changes the duty cycle for an open channel.
+        /// </summary>
+        /// <param name="pwmChip">The PWM chip.</param>
+        /// <param name="pwmChannel">The PWM channel.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage to change.</param>
+        public void ChangeDutyCycle(int pwmChip, int pwmChannel, double dutyCyclePercentage)
+        {
+            if (!_openChannels.Contains((pwmChip, pwmChannel)))
+            {
+                throw new InvalidOperationException("Can not change the duty cycle of a channel that is not open.");
+            }
+            if (dutyCyclePercentage < 0.0 || dutyCyclePercentage > 100.0)
+            {
+                throw new ArgumentException("Duty cycle must be a percentage in the range of 0-100.", nameof(dutyCyclePercentage));
+            }
+            _driver.ChangeDutyCycle(pwmChip, pwmChannel, dutyCyclePercentage);
         }
 
         /// <summary>
-        /// Changes the duty cycle.
+        /// Starts writing to an open channel.
         /// </summary>
         /// <param name="pwmChip">The PWM chip.</param>
         /// <param name="pwmChannel">The PWM channel.</param>
-        /// <param name="dutyCyclePercentage">The duty cycle percentage, from 0.0 to 100.</param>
-        /// <exception cref="InvalidOperationException">Can not change dutycycle to a pwm channel that is not yet opened.</exception>

```

Comment:
Why did we remove the exception comments? seems like they would be useful.

Reply:
I was questioning this myself, but it was the only couple of methods that included exception notes throughout API (based on my review so far).  I can add back if needed.

Reply:
I would probably suggest to simply not remove them from here, and then we can just log an issue so that we do a global pass to add all of the missing exception comments on methods that requires them.